jupyter/d2l-java/chapter_multilayer-perceptrons/weight-decay.ipynb

```diff
@@ -509,7 +506,70 @@
      "n": "5"
     }
    },
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "l1 norm of w: ND: () cpu() float32\n",
+      "152.8966\n",
+      "\n"
+     ]
+    },
+    {
+     "data": {
+      "text/html": [
+       "<div id=\"ee1fa0e5367f49aa8a32c4adcd33acf3\"></div>\n",
+       "<script>require(['https://cdn.plot.ly/plotly-1.44.4.min.js'], Plotly => {\n",
+       "var target_ee1fa0e5367f49aa8a32c4adcd33acf3 = document.getElementById('ee1fa0e5367f49aa8a32c4adcd33acf3');\n",
+       "var layout = {\n",
+       "    height: 600,\n",
+       "    width: 800,\n",
+       "    showlegend: true,\n",
+       "    xaxis: {\n",
+       "    title: 'epochCount',\n",
+       "    },\n",
+       "\n",
+       "    yaxis: {\n",
+       "    title: 'loss',\n",
+       "    },\n",
+       "\n",
+       "\n",
+       "};\n",
+       "\n",
+       "var trace0 =\n",
+       "{\n",
+       "x: ['5.0','10.0','15.0','20.0','25.0','30.0','35.0','40.0','45.0','50.0','55.0','60.0','65.0','70.0','75.0','80.0','85.0','90.0','95.0','100.0'],\n",
+       "y: [64.98075866699219, 62.144893646240234, 61.07378005981445, 60.646636962890625, 60.45595169067383, 60.35873031616211, 60.304893493652344, 60.272071838378906, 60.25064468383789, 60.23603439331055, 60.22640609741211, 60.219417572021484, 60.21455383300781, 60.21122741699219, 60.208858489990234, 60.20719909667969, 60.206031799316406, 60.205238342285156, 60.204673767089844, 60.20427703857422],\n",
+       "showlegend: true,\n",
+       "mode: 'lines',\n",
+       "type: 'scatter',\n",
+       "name: 'test',\n",
+       "};\n",
+       "var trace1 =\n",
+       "{\n",
+       "x: ['5.0','10.0','15.0','20.0','25.0','30.0','35.0','40.0','45.0','50.0','55.0','60.0','65.0','70.0','75.0','80.0','85.0','90.0','95.0','100.0'],\n",
+       "y: [31.32638931274414, 10.128734588623047, 3.565922975540161, 1.314842939376831, 0.4990929961204529, 0.1927323043346405, 0.07553799450397491, 0.0299958698451519, 0.012057898566126823, 0.004891997668892145, 0.0020053251646459103, 8.308241376653314E-4, 3.4801632864400744E-4, 1.476815959904343E-4, 6.333803321467713E-5, 2.7459678676677868E-5, 1.2073834113834891E-5, 5.383925781643484E-6, 2.432500650684233E-6, 1.1141642062284518E-6],\n",
+       "showlegend: true,\n",
+       "mode: 'lines',\n",
+       "type: 'scatter',\n",
+       "name: 'train',\n",
+       "};\n",
+       "\n",
+       "\n",
+       "var data = [ trace0, trace1];\n",
+       "Plotly.newPlot(target_ee1fa0e5367f49aa8a32c4adcd33acf3, data, layout);\n",
+       "})</script>\n"
+      ],
+      "text/plain": [
+       "tech.tablesaw.plotly.components.Figure@42b67112"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
    "source": [
     "train(0f);\n",
```

Comment:
The plotted graph pattern matching, but not as the same scale as gluon did

Reply:
I just ran it again and it does match very closely. Yes, it doesn't look like so because the python used 10^1 and 10^2 on the y-axis where I have a typical 0-100 one. So it doesn't look the same but the values match quite closely.

Reply:
This one is different, gluon is like 10^-1, where as the one you have is >1